projects/extension/ai/utils.py

```diff
@@ -7,3 +10,21 @@ def get_guc_value(plpy, setting: str, default: str) -> str:
     if val is None:
         val = default
     return val
+
+
+class VerboseTrace:
```

Comment:
If only used for requests, I would suggest we rename the class.
```suggestion
class VerboseRequestTrace:
```

Otherwise, change the message in

```
self.plpy.info(f"sending a request for {self.name}...")
```